Source/dom/export.js

```diff
@@ -84,12 +102,15 @@ function exportToImageFile(nativeObjects, options) {
     options
   )
 
-  // export the pages
-  pages.forEach(exporter.exportPage)
-
-  // export the layers
-  if (layers.length) {
-    exporter.exportLayers(layers)
+  try {
```

Comment:
I'm not too sure about wrapping everything in try/catch. That breaks any optimizations the compiler can make.

I'd rather explicitly check for it:
```js
if (!nativeObjects.parentPage()) {
  throw error
}
```

Reply:
I started going that route but then realized there are some edge cases in there. For example, some cases don't need a parent. This seemed less likely to cause conflicts if the internals were to change (except the error messages of course but that is covered by the tests). 

Any idea what that difference in optimizations would be? Significant for this?